arangod/Replication2/StateMachines/Document/MaintenanceActionExecutor.cpp

```diff
@@ -105,12 +106,13 @@ auto MaintenanceActionExecutor::executeModifyCollectionAction(
 auto MaintenanceActionExecutor::executeCreateIndex(
     ShardID shard, std::shared_ptr<VPackBuilder> const& properties,
     std::shared_ptr<methods::Indexes::ProgressTracker> progress) -> Result {
-  DatabaseGuard guard(_vocbase);
```

Comment:
You are loosing the Guard here.
This guard makes sure the vocbase pointer is not dropped.

Is there anything else that guarantees this guard?

Reply:
Come to think about it, should the guard actually be a member of the class that is only erased after the action is completed?